File: src/dataset/go/anagram_Exercism_go_rootulp/gemini/GeminiFlash_anagram_v1.go
package anagram

import (
	"strings"
)

func Detect(subject string, candidates []string) (anagrams []string) {
	subjectLower := strings.ToLower(subject)
	subjectCounts := getOccurrences(subjectLower)
	subjectLen := len(subjectLower)

	for _, candidate := range candidates {
		candidateLower := strings.ToLower(candidate)
		if len(candidateLower) != subjectLen || strings.EqualFold(subject, candidate) {
			continue
		}

		candidateCounts := getOccurrences(candidateLower)

		if areMapsEqual(subjectCounts, candidateCounts) {
			anagrams = append(anagrams, candidate)
		}
	}
	return anagrams
}

func getOccurrences(s string) map[rune]int {
	counts := make(map[rune]int, len(s))
	for _, r := range s {
		counts[r]++
	}
	return counts
}

func areMapsEqual(map1, map2 map[rune]int) bool {
	if len(map1) != len(map2) {
		return false
	}
	for key, val1 := range map1 {
		if val2, ok := map2[key]; !ok || val1 != val2 {
			return false
		}
	}
	return true
}

File: src/dataset/python/anagram/gemini/GeminiFlash_anagram_v1.py
def find_anagrams(word, candidates):
    word_lower = word.lower()
    word_counts = {}
    for char in word_lower:
        word_counts[char] = word_counts.get(char, 0) + 1

    anagrams = []
    for candidate in candidates:
        candidate_lower = candidate.lower()
        if len(word_lower) != len(candidate_lower) or word_lower == candidate_lower:
            continue

        candidate_counts = {}
        for char in candidate_lower:
            candidate_counts[char] = candidate_counts.get(char, 0) + 1

        if candidate_counts == word_counts:
            anagrams.append(candidate)

    return anagrams